### PR TITLE
aggregated annotations and labels.

### PR DESCRIPTION
**Pull request by zyjhtangtang**
#### What this PR does / why we need it:

Aggregate labels and annotations from poolservices status to service.

#### Does this PR introduce a user-facing change?
```release-note
YES
```

### CODE REVIEW

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	}
+	for key, value := range update {
+		m[key] = value
+	}
+
+	for key := range deletion {
+		delete(m, key)
+	}
+	return m
+}
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go:140]: diffMap and updateMap can be optimized as one function:

```
func mergeDesiredMap(currentMap, desiredMap map[string]string) bool {
    if currentMap == nil {
        currentMap = make(map[string]string)
    }

    modified := false
    for key, value := range desiredMap {
        if currentValue, exists := currentMap[key]; !exists || currentValue != value {
            currentMap[key] = value
            modified = true
        }
    }

    for key := range currentMap {
        if _, exists := desiredMap[key]; !exists {
            delete(currentMap, key)
            modified = true
        }
    }

    return modified
}
```

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/aggregatetoannotations.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+func aggregatePoolServicesAnnotations(poolServices []netv1alpha1.PoolService) map[string]string {
+	aggregatedAnnotations := make(map[string]string)
+	for _, ps := range poolServices {
+		aggregatedAnnotations = mergeMap(aggregatedAnnotations, ps.Status.AggregateToAnnotations)
+	}
+
+	return addAggregatePrefix(aggregatedAnnotations)
+}
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/aggregatetoannotations.go:34]: refer to aggregatePoolServicesLabels and optimize it.

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go (hunk 1 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/aggregatetolabels.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/openyurtio/openyurt/pkg/apis/network/v1alpha1"
+)
+
+func aggregatePoolServicesLabels(poolServices []v1alpha1.PoolService) map[string]string {
+	aggregatedLabels := make(map[string]string)
+	for _, ps := range poolServices {
+		aggregatedLabels = mergeMap(aggregatedLabels, ps.Status.AggregateToLabels)
+	}
+	return addAggregatePrefix(aggregatedLabels)
+}
```
[the remark this answers is not in the document]
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/aggregatetolabels.go:34]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/aggregatetoannotations.go (hunk 2 above)
Comment (zyjhtangtang) [reply pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/aggregatetoannotations.go:34]: done

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	for key, value := range currentMap {
+		if _, exist := result[key]; exist {
+			continue
+		}
+		if strings.HasPrefix(key, aggregateKeyPrefix) {
+			continue
+		}
+		result[key] = value
+	}
+	return result
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go:85]: result := make(map[string]string)

for key, value := range currentMap {
    if !strings.HasPrefix(key, aggregateKeyPrefix) {
         result[key] = value
    }
}

for key, value := range desiredMap {
    result[key] = value
}

File: pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+	for key, value := range m {
+		if !strings.HasPrefix(key, aggregateKeyPrefix) {
+			continue
+		}
+		newMap[key] = value
+	}
+	return newMap
+}
+
+func mergeDesireMap(currentMap, desiredMap map[string]string) map[string]string {
```
Comment (rambohe-ch) [inline pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go:69]: mergeDesireMap --> mergeAndPurgeMap

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**